application/deed/searchdeed/views.py

```diff
@@ -68,27 +68,67 @@ def show_authentication_code_page():
     return render_page
 
 
-def verify_auth_code(auth_code):
+@searchdeed.route('/confirming-mortgage-deed', methods=['POST'])
+def show_confirming_deed_page():
+
+    auth_code = request.form['auth_code']
 
     if auth_code is None or auth_code == '':
         return render_template('authentication-code.html', error=True)
 
-    deed_api_client = getattr(searchdeed, 'deed_api_client')
-    response = deed_api_client.verify_auth_code(str(session.get('deed_token')),
-                                                str(session.get('borrower_token')),
-                                                auth_code)
+    page_render = render_template('confirming-mortgage-deed.html', auth_code=request.form['auth_code'])
+    return page_render
 
-    if response.status_code == status.HTTP_200_OK:
-        return redirect(url_for('searchdeed.show_final_page'), code=307)
-    elif response.status_code == status.HTTP_401_UNAUTHORIZED:
-        return render_template('authentication-code.html', error=True)
-    else:
-        session['code-sent'] = None
-        session['service_timeout_at_send_code'] = None
+
+@searchdeed.route('/verify-auth-code', methods=['POST'])
+def verify_auth_code(auth_code=None):
+    if request.form['auth_code']:
+        auth_code = request.form['auth_code']
+
+    try:
+        deed_api_client = getattr(searchdeed, 'deed_api_client')
+        response = deed_api_client.verify_auth_code(str(session.get('deed_token')),
+                                                    str(session.get('borrower_token')),
+                                                    auth_code)
+
+        if response.status_code == status.HTTP_401_UNAUTHORIZED:
+            return jsonify({'error': True, 'redirect': url_for('authentication-code.html', error=True)})
+        elif response.status_code == status.HTTP_503_SERVICE_UNAVAILABLE or response.status_code \
+                == status.HTTP_500_INTERNAL_SERVER_ERROR:
+            return jsonify({'error': True, 'redirect': url_for('searchdeed.show_internal_server_error_page')})
+
+        return jsonify({'error': False})
+    except:
         session['service_timeout_at_verify_code'] = True
         raise exceptions.ServiceUnavailable
 
 
+@searchdeed.route('/verify-auth-code-no-js', methods=['POST'])
+def verify_auth_code_no_js():
+    auth_code = request.form['auth_code']
+
+    if auth_code is None or auth_code == '':
+        return redirect(url_for('searchdeed.show_authentication_code_page', error=True))
+
+    response = verify_auth_code(auth_code)
+
+    if response.status_code == status.HTTP_401_UNAUTHORIZED:
+        return redirect(url_for('authentication-code.html', error=True))
+    elif response.status_code == status.HTTP_503_SERVICE_UNAVAILABLE or response.status_code \
```

Comment:
I know its picky but if you're going to put code onto a continuation line i feel it would be better to put the line break in a more readable place such as after the or. placing it in the middle of an expression makes it harder to read